src/mpl/include/mpl_gpu.h

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPL_GPU_H_INCLUDED
+#define MPL_GPU_H_INCLUDED
+
+#include "mplconfig.h"
+
+#ifdef MPL_HAVE_GPU
```

Comment:
Instead of using ifdefs, it might be better to always build the functionality, but the query can always return "CPU pointer" if GPU support is not enabled.

Reply:
It has a fallback implementation to handle that.